File: Python/8kyu/8kyu-what's-the-real-floor.py
# What's the real floor?
# Language: Python
# Version: 3.10
# Difficulty: 8 kyu

def get_real_floor(n):
    if n > 0 and n < 13:
        return n - 1
    elif n >= 13:
        return n - 2
    else:
        return n


# Tests
import codewars_test as test
from solution import get_real_floor

@test.describe("Fixed Tests")
def fixed_tests():
    @test.it('Basic Test Cases')
    def basic_test_cases():
        test.assert_equals(get_real_floor(1), 0)
        test.assert_equals(get_real_floor(5), 4)
        test.assert_equals(get_real_floor(15), 13)

File: R/8kyu/8kyu-what's-the-real-floor.r
# What's the real floor?
# Language: R
# Version: 3.4.1
# Difficulty: 8 kyu

# {dplyr} solution
library(dplyr)
get_real_floor <- function(n) {
  case_when(n > 0 & n < 13 ~ n - 1,
            n >= 13 ~ n - 2,
            TRUE ~ n)
}

# Base R solution
get_real_floor <- function(n) {
  if (n > 0 & n < 13) {
    n - 1
  }else if (n >= 13) {
    n - 2
  }else{
    n
  }
}

# Tests
test_that('Fixed tests', {
  expect_equal(get_real_floor(1), 0)
  expect_equal(get_real_floor(5), 4)
  expect_equal(get_real_floor(15), 13) 
})